### ISSUE TITLE
feat: LSTM/GRU regime predictor for 1-min returns

### ISSUE THREAD

**Issue by SantiBarrios2002** [SantiBarrios2002/hmm-momentum-trading#80]
## Context

Compare a sequence-to-sequence deep learning model against the HMM/IOHMM. The LSTM/GRU can learn non-linear temporal dependencies that the HMM's Markovian assumption cannot capture.

### `src/ml/lstm.py`

**Model architecture:**
- Input: sliding window of features (from #79), shape `(batch, seq_len, D)`
- LSTM or GRU layers (1-2 layers, hidden_dim configurable)
- Output head: predict next-step return sign (classification) or value (regression)
- Lightweight — this is a comparison baseline, not a production model

**Training:**
- PyTorch (add to requirements)
- MSE loss for regression, or BCE for sign classification
- Adam optimizer, learning rate scheduler
- Early stopping on validation loss
- Walk-forward: train on expanding window, predict next day

## Functions

```python
class ReturnPredictor(nn.Module):
    """LSTM/GRU model for 1-min return prediction."""

def train_lstm(X_train, y_train, X_val, y_val, 
               hidden_dim=64, n_layers=1, seq_len=60,
               lr=1e-3, epochs=50, batch_size=256):
    """Train LSTM on feature sequences. Returns trained model + history."""

def predict_lstm(model, X_test, seq_len=60):
    """Generate predictions from trained LSTM. Returns (T,) predictions."""

def lstm_to_signal(predictions, threshold=0.0):
    """Convert return predictions to trading signal ∈ {-1, 0, +1}."""
```

## Tests (`tests/test_lstm.py`)

- [ ] Model forward pass produces correct output shape
- [ ] Training loss decreases over epochs on synthetic data
- [ ] Predictions are finite and bounded
- [ ] Signal conversion produces valid {-1, 0, +1} values
- [ ] Model handles variable sequence lengths

## Acceptance criteria

- [ ] Docstrings explaining architecture choices
- [ ] All tests pass
- [ ] Trains on 400k 1-min bars in reasonable time (< 10 min on CPU)
- [ ] Compatible with backtest framework (`src/strategy/backtest.py`)

## Tech stack note

This requires adding `torch` to the project dependencies. The CLAUDE.md "no pytorch" rule was for HMM core — ML baselines need ML frameworks.

## Dependencies

- #79 (ML feature engineering pipeline)
